homeassistant/components/imap/coordinator.py

```diff
@@ -80,21 +79,20 @@ async def idle_loop(self) -> None:
         """Wait for data pushed from server."""
         while True:
             try:
-                idle = await self.imap_client.idle_start()
+                idle: asyncio.Future = await self.imap_client.idle_start()
                 await self.imap_client.wait_server_push()
                 self.imap_client.idle_done()
-                async with async_timeout.timeout(10):
-                    await idle
-                await self.async_request_refresh()
+                await asyncio.wait_for(idle, 30)
```

Comment:
We prefer using `async_timeout.timeout`. Why did we change?

Reply:
Sorry for that. I was trying to figure out the cause of the timeout error that was happening. It turned out unrelated to this after all. I added a comment here(https://github.com/home-assistant/core/pull/74623#discussion_r1057207695) for the timeout issue.